packages/notebook-extension/src/index.ts

```diff
@@ -163,9 +163,15 @@ const closeTab: JupyterFrontEndPlugin<void> = {
     commands.addCommand(id, {
       label: trans.__('Close and Shut Down Notebook'),
       execute: async () => {
-        // Shut the kernel down, without confirmation
-        await commands.execute('notebook:shutdown-kernel', { activate: false });
-        window.close();
+        const confirm = window.confirm(
```

Comment:
Thanks @sai-chand04 for working on this.

As discussed in https://github.com/jupyter/notebook/issues/7398, the idea was to introduce a setting to let users override the default behavior. Which does not seem to be the case here for now.